tests/Resource/CloudAccount/EndpointTest.php

```diff
@@ -333,4 +343,106 @@ public function testSetPhpVersion() {
           ->setPhpVersion($entity, '7.2');
       });
   }
+
+  /**
+   * @covers Endpoint::createBackup
+   */
+  public function testCreateBackup() {
+    // custom request handler for sandbox
+    $request_handler = function ($request, $options) {
+      // check request path
+      $this->assertEquals('cloud-account/1/backup', $request->getUri()->getPath());
+
+
+      return new GuzzleResponse(
+        200,
+        ['Content-type' => 'application/json'],
+        $this->_getResource(static::_RESOURCE_NEW_BACKUP, false)
+      );
+    };
+
+    // kick off
+    $this->_getSandbox(null, $request_handler)
+      ->play(function ($api, $sandbox) {
+        $entity = $api->getModel(static::_SUBJECT_MODEL_FQCN)->set('id',1);
+        $endpoint = $api->getEndpoint(static::_SUBJECT_MODULE);
+        $results = $endpoint->createBackup($entity);
+        $this->assertEquals('filename.tgz',$results->get('filename'));
+        $this->assertEquals("123 MB",$results->get('filesize'));
+        $this->assertEquals(456,$results->get('filesize_bytes'));
+      });
+  }
+
+  /**
+   * @covers Endpoint::downloadBackup
+   */
+  public function testDownloadBackup() {
+    // custom request handler for sandbox
+    $assertionCounter = 0;
+    $request_handler = function ($request, $options) use (&$assertionCounter) {
+      // check request path
+
+      switch (++$assertionCounter) {
+        case 1:
+          $this->assertEquals('cloud-account/1/backup', $request->getUri()->getPath());
+          break;
+
+        case 2:
+          $this->assertEquals('/siteworx/index', $request->getUri()->getPath());
+          break;
+      }
+
+      return new GuzzleResponse(
+        200,
+        ['Content-type' => 'application/json'],
+        $this->_getResource(static::_RESOURCE_BACKUPS, false)
+      );
+    };
+
+    // kick off
+    $this->_getSandbox(null, $request_handler)
+      ->play(function ($api, $sandbox) {
+        $filename = 'filename.tgz';
+        $path = "/tmp";
+
+        $entity = $api->getModel(static::_SUBJECT_MODEL_FQCN)->set('id',1);
+        $endpoint = $api->getEndpoint(static::_SUBJECT_MODULE);
+        $endpoint->downloadBackup($entity, $filename, $path);
+
+        $path = trim($path);
+        if (substr($path, -1) !== DIRECTORY_SEPARATOR) {
+          $path .= DIRECTORY_SEPARATOR;
+        }
+
+        $full_filename = $path . $filename;
+        $this->assertTrue(file_exists($full_filename));
```

Comment:
dislike tests writing to the filesystem. two ways we could handle this:
- use something like [vfsStream](https://medium.com/weebly-engineering/phpunit-mocking-the-file-system-using-vfsstream-5d7d79b1eb2a). I've never actually tried this but it seems convenient/transparent.
- create a function `Nexcess\Sdk\CloudAccount\Backup\file_put_contents` that does assertions, but doesn't touch the filesystem. I've done this before and it works out well.

Reply:
long way to go for something like this. As long as they cleanup after themselves, I don't see a problem. The only real problem I can see is that this would prob fail on a windows machine. maybe path should be "./"

Reply:
also, same could be said about the resource files. vfsStream seems to be able to replace them as well. If we are going to remove the file system as a dependency, let's do it. I'm not in favor of it, but if we are going to do it, let's do it.

Reply:
vfs works beautifully. 

I do still stand by the idea that all resource files should be included this way. I dislike littering the fs with resource directories. This would allow us to put them all in one class and get to them anywhere.